src/redux/reducers/notificationReducer.ts

```diff
@@ -0,0 +1,65 @@
+import uuid from 'uuid';
+
+import i18n from '../../translations/i18n';
+import { Notification } from '../../types/notification.types';
+import { AuthActions, LOGIN_FAILURE, REFRESH_TOKEN_FAILURE } from '../actions/authActions';
+import {
+  ADD_NOTIFICATION,
+  CLEAR_NOTIFICATIONS,
+  NotificationActions,
+  REMOVE_NOTIFICATION,
+} from '../actions/notificationActions';
+import { ORCID_REQUEST_FAILURE, ORCID_SIGNIN_FAILURE, OrcidActions } from '../actions/orcidActions';
+import {
+  CREATE_RESOURCE_FAILURE,
+  CREATE_RESOURCE_SUCCESS,
+  GET_RESOURCE_FAILURE,
+  ResourceActions,
+  UPDATE_RESOURCE_FAILURE,
+  UPDATE_RESOURCE_SUCCESS,
+} from '../actions/resourceActions';
+import { SEARCH_FAILURE, SearchActions } from '../actions/searchActions';
+import { SET_USER_FAILURE, SET_USER_SUCCESS, UserActions } from '../actions/userActions';
+
+export const notificationReducer = (
+  state: Notification[] = [],
+  action:
+    | NotificationActions
+    | AuthActions
+    | OrcidActions
+    | UserActions
+    | NotificationActions
+    | SearchActions
+    | ResourceActions
+) => {
+  switch (action.type) {
+    case ORCID_SIGNIN_FAILURE:
+    case ORCID_REQUEST_FAILURE:
+    case REFRESH_TOKEN_FAILURE:
+    case SET_USER_FAILURE:
+    case SEARCH_FAILURE:
+    case CREATE_RESOURCE_FAILURE:
+    case CREATE_RESOURCE_SUCCESS:
+    case UPDATE_RESOURCE_FAILURE:
+    case UPDATE_RESOURCE_SUCCESS:
+    case GET_RESOURCE_FAILURE:
+    case LOGIN_FAILURE:
+      return [...state, { key: uuid.v4(), ...action }];
```

Comment:
Her vil vel `key` overskrives av eventuell `key` i `action`. Er dette meningen? (Samme under)

Reply:
også: hva med å rename til `id`? Jeg får litt assosiasjoner til at denne bare brukes som key for rendring i react, men mulig det bare er meg

Reply:
key finnes ikke på de actionene der, så det blir lagt til. kan godt rename til id 👍

Reply:
grunnen til at det er key er fordi på options-objektet til notifications så heter det key